src/main/java/com/ebivariation/contigalias/dus/AssemblyReportReader.java

```diff
@@ -0,0 +1,142 @@
+        if (tagEnd == -1) {
+            return;
+        }
+        String tag = line.substring(2, tagEnd);
+        String tagData = line.substring(tagEnd + 1).trim();
+        switch (tag) {
+            case "Assembly name": {
+                assemblyEntity.setName(tagData);
+                break;
+            }
+            case "Organism name": {
+                assemblyEntity.setOrganism(tagData);
+                break;
+            }
+            case "Taxid": {
+                assemblyEntity.setTaxid(Long.parseLong(tagData));
+                break;
+            }
+            case "GenBank assembly accession": {
+                assemblyEntity.setGenbank(tagData);
+                break;
+            }
+            case "RefSeq assembly accession": {
+                assemblyEntity.setRefseq(tagData);
+                break;
+            }
+        }
+    }
+
+    private void parseChromosomeLine(String line) {
+
+        ChromosomeEntity chromosomeEntity = new ChromosomeEntity();
+
+        int tabIndex = line.indexOf('\t');
+        if (tabIndex != -1) {
+            String sequaenceName = line.substring(0, tabIndex);
+            chromosomeEntity.setName(sequaenceName);
+            line = line.substring(tabIndex + 1);
+        }
+
+        tabIndex = line.indexOf('\t');
+        if (tabIndex != -1) {
+            line = line.substring(tabIndex + 1);
+        }
+
+        tabIndex = line.indexOf('\t');
+        if (tabIndex != -1) {
+            line = line.substring(tabIndex + 1);
+        }
+
+        tabIndex = line.indexOf('\t');
+        if (tabIndex != -1) {
+            String assignedModeculeLocationType = line.substring(0, tabIndex);
+            if (!assignedModeculeLocationType.equals("Chromosome")) {
+                return;
+            }
+            line = line.substring(tabIndex + 1);
+        }
+
+        tabIndex = line.indexOf('\t');
+        if (tabIndex != -1) {
+            String genbankAccn = line.substring(0, tabIndex);
+            chromosomeEntity.setGenbank(genbankAccn);
+            line = line.substring(tabIndex + 1);
+        }
+
+        tabIndex = line.indexOf('\t');
+        if (tabIndex != -1) {
+            line = line.substring(tabIndex + 1);
+        }
+
+        tabIndex = line.indexOf('\t');
+        if (tabIndex != -1) {
+            String refseqAccn = line.substring(0, tabIndex);
+            chromosomeEntity.setRefseq(refseqAccn);
+        }
+
+        chromosomeEntity.setAssembly(this.assemblyEntity);
+        if (assemblyEntity == null) {
+            assemblyEntity = new AssemblyEntity();
+        }
+        List<ChromosomeEntity> chromosomes = this.assemblyEntity.getChromosomes();
+        if (chromosomes == null) {
+            chromosomes = new LinkedList<>();
+            assemblyEntity.setChromosomes(chromosomes);
+        }
+        chromosomes.add(chromosomeEntity);
+
+    }
+
+    public long getLineCount() {
+        return reader.lines().count();
```

Comment:
I think the `lines()` method consumes the reader, so if you have an AssemblyReportReader you can either parse the asm report *or* count its lines, and this constraint is not reflected in the class interface.

Reply:
How can I fix this? Currently this is only used to check if reader has loaded the data. I can just modify the test to use reader.ready() and remove this method entirely.

Reply:
yes, that sounds good. use ready and remove the method.